src/core/fonts.js

```diff
@@ -159,6 +159,7 @@ function adjustToUnicode(properties, builtInEncoding) {
 
 class Glyph {
   constructor(
+    originalCharCode,
```

Comment:
Don't you need to update the `matchesForCache` method and its call-site as well, to avoid inconsistent data?
 - https://github.com/mozilla/pdf.js/blob/425e58431a354673817f4687b6cf27e7350c4945/src/core/fonts.js#L181
 - https://github.com/mozilla/pdf.js/blob/425e58431a354673817f4687b6cf27e7350c4945/src/core/fonts.js#L2958